ksql-engine/src/main/java/io/confluent/ksql/QueryEngine.java

```diff
@@ -115,9 +115,10 @@ private PlanNode buildQueryLogicalPlan(String sqlExpression, final Query query,
                          ? ksqlStructuredDataOutputNode.getTheSourceNode().getTimestampField()
                          : ksqlStructuredDataOutputNode.getTimestampField(),
                          ksqlStructuredDataOutputNode.getKsqlTopic());
-
-      tempMetaStore.putTopic(ksqlStructuredDataOutputNode.getKsqlTopic());
-      tempMetaStore.putSource(structuredDataSource.cloneWithTimeKeyColumns());
+      if (analysis.isDoCreateInto()) {
```

Comment:
What does this mean? Is there another name that might be more descriptive?

Reply:
i still think a different name might be better, i.e, `shouldUpdateMetastore` or `shouldCreateEntity` or similar